### DISCUSSION TITLE
Folder Template triggers when it shouldn't

### DISCUSSION THREAD

**Discussion by zanodor** [SilentVoid13/Templater#1122] [Help] [answered]
On iOS, I am using a [bookmarklet](https://gist.github.com/kepano/90c05f162c37cf730abb8ff027987ca3) that I configured to save the URL contents to an inbox folder. 
My folder template should trigger on new files created in the root folder (`/`).

Probably due to the fact the Obsidian is opened through Obsidian URI, Templater detects the new file as a newly created (empty?) file with a filename "inbox/filename.md". The bookmarklet-saved file with content gets overwritten with a file (with the same filename) with pre-configured YAML and empty content.  

Current workaround:
- Disable the folder template or the whole plugin before using the bookmarklet. 
- Keep Templater functions enabled and when using the bookmarklet, at the time of Obsidian coming into focus and the file appearing, press enter quickly (even twice) (the cursor is always on the name of the file created, not in the main body text) before Templater could overwrite the file.  
- What I use now: let Templater wreck my file and then I just hit undo (not all undo's work: I use the Obsidian command 'undo' I put on the mobile toolbar).  

## Further examination

Actually, as I found out if I create a file in the way I usually do (click on recent files and enter the filename or click on a folder in the File Tree Alternative plugin to create a file in the specified folder), the folder template will still trigger.  
- Just tried it on the desktop as well; it does so there as well. I don't remember this from before.   

## Question

Is there a way to use an exclude under root folder? Or a regular expression?

Is this happening to anybody else?

**Comment by Zachatoo** [comment]
If you're saying that you have a folder template setup to trigger at the root of your vault (`/`) and you have a nested folder that does not have a folder template setup for it, and creating a note in a nested folder is triggering the root folder template, then that is working as intended.

You can setup a folder template for your nested folder to override the root folder template so the root folder template doesn't execute.

Important! In the folder template settings, you'll want to make sure the order of your folder templates is such that the root folder template is below your nested folder template, so the nested folder template runs and prevents the root folder template from running.

**Reply by zanodor** [reply]
Thanks for taking the time to answer.

> Important! In the folder template settings, you'll want to make sure the order of your folder templates is such that the root folder template is below your nested folder template, so the nested folder template runs and prevents the root folder template from running.  
- Interesting, I did not know that.
Although now I have a different workflow on iOS and no longer have multiple folder templates, just the one.

> You can setup a folder template for your nested folder to override the root folder template so the root folder template doesn't execute.  
- I would like that, but how to achieve "do nothing"? I tried javascript (not my forte) but couldn't make it work. I want the markdownified text intact.

**Reply by Zachatoo** [reply]
> how to achieve "do nothing"?

I'm not totally sure, but I would try making a template that is empty and use that for the template for the folder template.

**Reply by danielo515** [reply]
The root template still triggers when creating a new note from template, but it should not. I think there is an issue about that

**Comment by zanodor** [comment, marked as answer]
In the end, I managed to solve it. Same folder template, no inbox folder or folder template used.  

In the bookmarklet, I added a prefix "webclip - " to the filename.  
In the Meta Templater [method](https://m.youtube.com/watch?v=5zcdG6ZWja4) used, I added the prefix to handle this special case differently, then I used `<% tp.file.content %>` for the template triggered when prefix is `webclip - `, and now the file content stays intact.  

I tried using a simple cursor placement method which was not enough.  

In the end, I decided to delete the prefix as well (the JavaScript snippet must be downloaded from Pamela's Github (see YT video's description), and extra lines must be added to the Meta Templater "master" template for this to work):    

This template is called when the bookmarklet-generated markdown file is created:  
````
<%-* 
var cleanTitle = tp.user.getTitleSnippet(tp.file.title)
var title = `${cleanTitle}`;
await tp.file.rename(`${title}`)
_%>
<% tp.file.content %>
````
